### ISSUE TITLE
S3Exception MalFormed : akka.stream.alpakka.s3.S3Exception:  (Code: , RequestID: , HostID: )

### ISSUE THREAD

**Issue by GhaileneSouissi** [akka/alpakka#2310]
Akka version:
```
"com.typesafe.akka" %% "akka-http" % "10.1.10",
"com.typesafe.akka" %% "akka-stream" % "2.5.25",
"com.typesafe.akka" %% "akka-http-xml" % "10.1.10"
```

Alpakka version :
`"com.lightbend.akka" %% "akka-stream-alpakka-s3" % "2.0.0"`

Hello, I am using alpakka S3 to upload files into amazon S3 storage system. The implementation I made is simple :

```
 def s3Settings(implicit system: ActorSystem) = S3Ext.get(system).settings
    .withCredentialsProvider(credentielProvider)
    .withS3RegionProvider(regionProvider)
    .withBufferType(MemoryBufferType)
    .withListBucketApiVersion(ListBucketVersion2)
    .withForwardProxy(ForwardProxy("endpoint",Port , Some(ForwardProxyCredentials("username", "password"))))
    .withEndpointUrl(endpointUrl)
    .withPathStyleAccess(true)
```

I am using docker to simulate an S3 system (minio) and everything worked just fine.

After adding a proxy and changing the url to connect to my S3 system in production,  it works when i download files and delete files from S3. but when i try to upload files, it throws this exception :

`akka.stream.alpakka.s3.S3Exception: (Code: , RequestID: , HostID: )`

Is there any way to have more information in the exception ?

**Comment by ennru** [comment]
Looking at `S3Exception` I get the impression it will never parse the returned error correctly:

https://github.com/akka/alpakka/blob/5534c616c3c654d26b7f535287943bd1d38b398b/s3/src/main/scala/akka/stream/alpakka/s3/S3Exception.scala#L15-L33

Please suggest an improvement to this error handling.

**Comment by ihrimech** [comment]
Hi @ennru 

I would like to contribute to this issue but I'll need some guidance since doing it for the first time. What exactly made you think that the parsing would never be made correctly ?

**Comment by ennru** [comment]
Thank you, I looked into it and created #2341.
Can you try it with your setup and see if it improves your situation, please?

**Comment by ennru** [comment]
Fixed with #2341
